dissect/target/plugins/os/windows/dpapi/dpapi.py

```diff
@@ -119,6 +139,19 @@ def decrypt_system_blob(self, data: bytes) -> bytes:
 
         return blob.clear_text
 
+    def decrypt_blob(self, data: bytes, mk_passwords: list) -> bytes:
+        """Attempt to decrypt the given bytes using any of the available master keys."""
+        blob = DPAPIBlob(data)
+        self.mk_passwords = mk_passwords
```

Comment:
Don't think this is a great idea. It's probably best to improve the keychain functionality, or otherwise design a better API for this. I have some ideas, but let's first explore the keychain route.